src/main/java/com/google/devtools/build/lib/runtime/UiOptions.java

```diff
@@ -46,57 +47,19 @@ public enum UseCurses {
   }
 
   /** Converter for {@link EventKind} filters * */
-  public static class EventFiltersConverter extends Converter.Contextless<List<EventKind>> {
-
-    /** A converter for event kinds. */
-    public static class EventKindConverter extends EnumConverter<EventKind> {
-
-      public EventKindConverter(String typeName) {
-        super(EventKind.class, typeName);
-      }
-    }
-
-    private final CommaSeparatedOptionListConverter delegate;
-
-    public EventFiltersConverter() {
-      this.delegate = new CommaSeparatedOptionListConverter();
-    }
+  public static class EventFiltersConverter extends CommaSeparatedOptionListConverter {
+    EnumConverter<EventKind> validator = new EnumConverter<>(EventKind.class, "ui event") {};
 
     @Override
-    public List<EventKind> convert(String input) throws OptionsParsingException {
+    public ImmutableList<String> convert(String input) throws OptionsParsingException {
```

Comment:
What is the reason that you moved the actual conversion out of `convert`?

Reply:
I'm not entirely sure as it's been a long time since I made this change, but I think the reason was that I needed to have the union of all the the inputs to compute the filters. Options with `allowMultiple = true` will aways concatenate the list converted values for each `input`, that is why it wasn't possible to add an event back in: once an `input` was converted to filter an EventKind, a subsequent `input` could not remove the filter from the accumulator. I found it easier to just move the conversion outside, after all the strings have been accumulated.

Reply:
@meisterT Rebased and moved all the conversions back into the converter. Need to store two separate lists of events (removed and added) thou. Let me know if there is something I'm missing to avoid this.